Buildkite: Run weeder as early as possible, after build, before test

rvl opened the pull request:
1. Moves the Weeder command earlier in the Stack Rebuild step.
2. Updates Weeder to 2.2.0, which fixes the problem of false weeds due to stale `.hie` files.

rvl committed:
Buildkite: Run weeder as early as possible, after build, before test

jonathanknowles commented on lib/core/src/Cardano/Wallet.hs:
```diff
@@ -2941,3 +2942,7 @@ instance HasSeverityAnnotation TxSubmitLog where
+
+-- | Empty fluff.
+nullFluff :: Monoid a => a
+nullFluff = mempty
```
Hi @rvl 

It seems this doesn't cause weeder to fail, at least not in [this build](https://buildkite.com/input-output-hk/cardano-wallet/builds/16364#24482922-1720-469b-9589-5a4f096483df).

Perhaps I've misunderstood something?

rvl replied:
Yes - https://github.com/input-output-hk/cardano-wallet/pull/2851#issuecomment-906917880

rvl committed:
nix-shell: Bump Weeder from 2.1.3 to 2.2.0

This release includes a fix for the issue with reading .hie files
which don't have a corresponding .hs file.

rvl committed:
Regenerate materialized nix

rvl committed:
Buildkite: Don't remove .hie files from previous builds

Weeder now knows how to ignore orphaned .hie files.

rvl committed:
Remove weeds from Shelley.Compatibility

rvl committed:
weeder.dhall: Ignore some weeds

rvl committed:
Buildkite: specify .stack-work directories on weeder command-line
